### PR TITLE
feat: add wrapper to compute gradient

### LINKED ISSUE #33: [Optimization] I tried to find the optimization using caputol1method
import numpy as np
import matplotlib.pyplot as plt
from pycaputo.utils import Array
from pycaputo import CaputoDerivative, CaputoL1Method, Side, diff
from pycaputo.grid import make_uniform_points
from pycaputo.utils import figure, set_recommended_matplotlib

# Define the function
def f(x: Array) -> Array:
    return (x - 5) ** 2

# Set up Caputo derivative and method
d = CaputoDerivative(order=0.9, side=Side.Left)
method = CaputoL1Method(d)

# Generate points for plotting and calculating numerical Caputo derivative
p = make_uniform_points(256, a=0.0, b=2.0)
df_num = np.array([diff(method, f, np.array([x]))[0] for x in p.x])

# Plot the Caputo derivative using the L1 method
set_recommended_matplotlib()
with figure("caputo-derivative-l1.svg") as fig:
    ax = fig.gca()
    ax.plot(p.x, df_num, lw=5, label="$L1~ Method$")
    ax.set_xlabel("$x$")
    ax.set_ylabel(r"$D^\alpha_C[f](x)$")
    ax.legend()
    plt.show()

# Initial setup for optimization using CaputoL1Method
x = 0
learning_rate = 0.1
error = []

plt.plot(p.x, df_num, lw=5, label="$L1~ Method$")
plt.xlabel('X')
plt.ylabel(r"$D^\alpha_C[f](x)$")

for i in range(25):
    # Use CaputoL1Method for optimization
    grad = diff(method, f, np.array([x]))[0]
    x = x - learning_rate * grad
    y = (x - 5) ** 2
    error.append(y)
    plt.scatter(x, y)

    # Print the values corresponding to each iteration
    print(x, grad)
    
plt.show()
I have some errors while finding the fractional gradient descent. Please help me to solve this issue.
Thank you.

### PR DESCRIPTION
**Pull request by alexfikl**
This attempts to add a gradient helper that just applies a method component by component. It probably needs more work, but it seems to work for a simple example.

Fixes #33

### CODE REVIEW

File: pycaputo/__init__.py [hunk 1]
```diff
@@ -51,14 +67,107 @@ def quad(
...
+
+    def make_component_p(i: tuple[int, ...]) -> Points:
+        return p.translate(a[i], x[i])
+
+    import pycaputo.differentiation as d
+
+    result = np.empty_like(x)
+    for i in np.ndindex(x.shape):
+        # FIXME: this should just compute the gradient at -1
+        result[i] = d.diff(m, make_component_f(i), make_component_p(i))[-1]
```
Comment (alexfikl) [inline pycaputo/__init__.py:168]: The logic for computing the gradient is just here. The rest of the code is just moving things around.
